<?xml version="1.0" ?>
<launch>
 
 # You can fill in here the name of your DS or pass it through the command line or another launch file!
 <arg name="DS_name"       default="Curve_go_right"/>
 <arg name="viz_DS_path"   default="true"/>
# Example Options:
  - Curve_go_left
  - Curve_go_right
  - free_ws
  - push_down
  - 3D-CShape-bottom
  - sink_seds
  - via-point_seds

# Things to Know!
# input_topic name should contain          -- geometry_msgs::Pose
# output_topic_name will write to          -- geometry_msgs::Twist
# output_orientation_topic_name will write -- geometry_msgs::Quaternion
<group ns="ds1">
    <param name="input_topic_name"                  value="/lwr/ee_pose"/>
    <param name="output_topic_name"                 value="/ds1/desired_velocity"/>
    <param name="output_filtered_topic_name"        value="/lwr/joint_controllers/passive_ds_command_vel"/>
    <param name="publish_DS_path"                   value="false"/>
    <rosparam command="load" file="$(find ds_motion_generator)/config/learned_DS/seDS/$(arg DS_name).yml"/>
    <node name="ds_node" pkg="ds_motion_generator"  type="seDS_motionGenerator_node" output="screen"/>
</group>

<group if="$(arg viz_DS_path)">
  <group ns="ds1_path">
      <param name="input_topic_name"                  value="/lwr/ee_pose"/>
      <param name="output_topic_name"                 value=""/>
      <param name="output_filtered_topic_name"        value=""/>    
      <param name="publish_DS_path"                   value="true"/>
      <rosparam command="load" file="$(find ds_motion_generator)/config/learned_DS/seDS/$(arg DS_name).yml"/>
      <node name="ds_node" pkg="ds_motion_generator"  type="seDS_motionGenerator_node" output="screen"/>
  </group>
</group>

# This node published the desired orientation at the target
<node pkg="rostopic" type="rostopic" name="angle_setting" args="pub /lwr/joint_controllers/passive_ds_command_orient geometry_msgs/Quaternion '{x:  0.0469, y: 0.9987, z: 0.0181, w:  -0.0015}' -r 100" />



</launch>
